pymc_marketing/clv/plotting.py

```diff
@@ -344,15 +346,138 @@ def plot_probability_alive_matrix(
         ylabel=ylabel,
         title=title,
     )
-    force_aspect(ax)
+    _force_aspect(ax)
 
     # plot colorbar beside matrix
     plt.colorbar(pcm, ax=ax)
 
     return ax
 
 
-def force_aspect(ax: plt.Axes, aspect=1):
+def plot_expected_purchases(
+    model,
+    purchase_history: pd.DataFrame,
+    customer_id_col: str,
+    datetime_col: str,
+    t: int,
+    plot_cumulative: bool = True,
+    t_cal: int | None = None,
+    datetime_format: str | None = None,
+    time_unit: str = "D",
+    time_scaler: float | None = 1,
+    sort_transactions: bool | None = True,
+    set_index_date: bool | None = False,
+    title: str | None = None,
+    xlabel: str = "Time Periods",
+    ylabel: str = "Purchases",
+    ax: plt.Axes | None = None,
+    **kwargs,
+):
+    """Plot actual and expected purchases over time for a fitted ``BetaGeoModel`` or ``ParetoNBDModel``.
+
+    This function is based on the formulation on page 8 of [1]_. Specifically, we take only customers who have made
+    their first transaction before the specified number of ``t`` time periods, and run
+    ``expected_purchases_new_customer()`` for all remaining time periods. Results can be either cumulative or
+    incremental.
+
+    Adapted from legacy ``lifetimes`` library:
+    https://github.com/CamDavidsonPilon/lifetimes/blob/master/lifetimes/plotting.py#L392
+
+    Parameters
+    ----------
+    model :
+        A fitted ``BetaGeoModel`` or ``ParetoNBDModel``.
+    purchase_history : ~pandas.DataFrame
+        A Pandas DataFrame containing *customer_id_col* and *datetime_col*.
+    customer_id_col : string
+        Column in the *transactions* DataFrame denoting the *customer_id*.
+    datetime_col :  string
+        Column in the *transactions* DataFrame denoting datetimes purchase were made.
+    t : int
+        Number of time units since earliest purchase to include in plot.
+    plot_cumulative : bool
+        Default: *True*
+        Plot cumulative purchases over time. Set to *False* to plot incremental purchases.
+    t_cal : int, optional
+        If testing model on unobserved data, specify number of time units in training data to add an indicator for
+        the start of the testing period.
```

Comment:
I am having a bit of a hard time understand the name `t_cal`. Could be the suffix `cal` throwing me off.

Reply:
Ah, a `lifetimes` leftover I overlooked. This marks the start of unobserved data if evaluating the model with `rfm_train_test_split`. I've renamed it to `t_unobserved`.